lib/datadog/tracing/contrib/action_pack/action_dispatch/instrumentation.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative '../../../metadata/ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module ActionPack
+        module ActionDispatch
+          # Instrumentation for ActionDispatch components
+          module Instrumentation
+            module_function
+
+            def set_http_route_tags(route_spec, script_name)
+              return unless Tracing.enabled?
+
+              return unless route_spec
+
+              request_trace = Tracing.active_trace
+              return unless request_trace
+
+              request_trace.set_tag(Tracing::Metadata::Ext::HTTP::TAG_ROUTE, route_spec.to_s.gsub(/\(.:format\)\z/, ''))
```

Comment:
Maybe add a note here why the format is removed from route? Because there can be other parameters in the rest of the route?

Reply:
I am actually not sure whether we want to remove it. A fully qualified rails route will have `(.:format)` at the end. The reason why I added this formatting is because it was in the [first attempt](https://github.com/DataDog/dd-trace-rb/pull/3345) to implement this feature.

Reply:
I think this decision is not mine to make therefore I will resolve this comment.